src/swap/utils.ts

```diff
@@ -1,243 +1,250 @@
-import {Algodv2} from "algosdk";
-
 import {CONTRACT_VERSION} from "../contract/constants";
-import {AssetWithIdAndAmount, TinymanAnalyticsApiAsset} from "../util/asset/assetModels";
-import {InitiatorSigner, SignerTransaction, SupportedNetwork} from "../util/commonTypes";
-import {PoolReserves, V1PoolInfo, V2PoolInfo} from "../util/pool/poolTypes";
-import {SwapQuoteWithPool, SwapQuote} from "./types";
+import {InitiatorSigner, SignerTransaction} from "../util/commonTypes";
+import {V1PoolInfo} from "../util/pool/poolTypes";
+import {
+  GetSwapQuoteBySwapTypeParams,
+  GenerateSwapTxnsParams,
+  GetSwapQuoteParams,
+  SwapQuote,
+  SwapQuoteType,
+  ExecuteSwapCommonParams
+} from "./types";
 import {SwapType} from "./constants";
 import {SwapV1_1} from "./v1_1";
 import {SwapV2} from "./v2";
-import {V1_1_SWAP_TOTAL_FEE} from "./v1_1/constants";
-import {getV2SwapTotalFee} from "./v2/util";
+import {
+  getBestQuote,
+  getSwapQuoteContractVersion,
+  isSwapQuoteErrorCausedByAmount
+} from "./common/utils";
+import {getAssetId} from "../util/asset/assetUtils";
 import {isPoolEmpty} from "../util/pool/common";
-import OutputAmountExceedsAvailableLiquidityError from "../util/error/OutputAmountExceedsAvailableLiquidityError";
+import SwapQuoteError, {SwapQuoteErrorType} from "../util/error/SwapQuoteError";
 
 /**
- * Gets quotes for swap from each pool passed as an argument,
- * and returns the best quote (with the highest rate).
+ * Gets the best quote for swap from the pools and swap router and returns the best option.
  */
-export function getQuote(params: {
-  type: SwapType;
-  pools: {info: V1PoolInfo | V2PoolInfo; reserves: PoolReserves}[];
-  assetIn: Pick<TinymanAnalyticsApiAsset, "id" | "decimals">;
-  assetOut: Pick<TinymanAnalyticsApiAsset, "id" | "decimals">;
-  amount: number | bigint;
-}): Promise<SwapQuoteWithPool> {
-  if (params.pools.every((pool) => isPoolEmpty(pool.reserves))) {
-    throw new Error("No pools available for swap");
+export function getQuote(params: GetSwapQuoteParams): Promise<SwapQuote> {
+  const {type, pools, isSwapRouterEnabled} = params;
+
+  if (!isSwapRouterEnabled && pools.every((pool) => isPoolEmpty(pool.reserves))) {
+    throw new SwapQuoteError(
+      SwapQuoteErrorType.NoAvailablePoolError,
```

Comment:
I guess on the web client we will check `SwapQuoteErrorType.NoAvailablePoolError && !isSwapRouterEnabled` and then display the "enable router" CTA. We can also make the error type specific but both is okay 
```suggestion
      SwapQuoteErrorType.NoAvailablePoolError_EnableRouter,
```